java/semantickernel-api/src/main/java/com/microsoft/semantickernel/Kernel.java

```diff
@@ -1,82 +1,92 @@
 // Copyright (c) Microsoft. All rights reserved.
 package com.microsoft.semantickernel;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-
-import javax.annotation.Nullable;
-
 import com.microsoft.semantickernel.builders.Buildable;
 import com.microsoft.semantickernel.builders.SemanticKernelBuilder;
 import com.microsoft.semantickernel.hooks.KernelHooks;
 import com.microsoft.semantickernel.orchestration.FunctionInvocation;
 import com.microsoft.semantickernel.orchestration.KernelFunction;
 import com.microsoft.semantickernel.plugin.KernelPlugin;
-import com.microsoft.semantickernel.plugin.KernelPluginCollection;
-import com.microsoft.semantickernel.semanticfunctions.PromptTemplate;
 import com.microsoft.semantickernel.services.AIServiceSelection;
 import com.microsoft.semantickernel.services.AIServiceSelector;
 import com.microsoft.semantickernel.services.OrderedAIServiceSelector;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 /**
  * Interface for the semantic kernel.
  */
 public class Kernel implements Buildable {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(Kernel.class);
+
     private final AIServiceSelector serviceSelector;
     private final KernelPluginCollection plugins;
     private final KernelHooks globalKernelHooks;
 
     public Kernel(
         AIServiceSelector serviceSelector,
-        @Nullable KernelPluginCollection plugins,
+        @Nullable List<KernelPlugin> plugins,
         @Nullable KernelHooks globalKernelHooks) {
         this.serviceSelector = serviceSelector;
-
         if (plugins != null) {
-            this.plugins = new KernelPluginCollection(plugins.getPlugins());
+            this.plugins = new KernelPluginCollection(plugins);
         } else {
             this.plugins = new KernelPluginCollection();
         }
 
         this.globalKernelHooks = new KernelHooks(globalKernelHooks);
     }
 
+    @SuppressWarnings({"rawtypes", "unchecked"})
     public <T> FunctionInvocation<T> invokeAsync(
         String pluginName,
         String functionName) {
-        KernelFunction function = plugins.getFunction(pluginName, functionName);
+        KernelFunction function = getFunction(pluginName, functionName);
         return invokeAsync(function);
     }
 
     public <T> FunctionInvocation<T> invokeAsync(KernelFunction<T> function) {
         return function.invokeAsync(this);
     }
 
+    public void addPlugin(KernelPlugin plugin) {
```

Comment:
should this return `Kernel`?

Reply:
To be consistent with fluent style, I guess so. I'll make this adjustment along with the javadoc.